### ISSUE TITLE
Deployment: npm does not pick up proxy and https-proxy settings

### ISSUE THREAD

**Issue by DonMartin76** [NREL/api-umbrella#188]
Hello,

I am currently struggling with getting the development setup (using vagrant) up and running. First thing was that the entire Vagrantfile does not work for me behind our corporate proxy. I managed to workaround that using the normal vagrant-proxy plugin, and the scripts continued for a while more. Actually, this also makes vagrant/chef install a /etc/environment containing the right values for http_proxy and https_proxy.

So far, so good, Enter `npm`. For whatever reason the node.js decided it is reasonable for `npm` not to pick up the system wide proxy settings, I can't seem to find the right place to add it to the scripts to make npm proxy aware inside the VM. I always end up with the following errors:

```
==> default: npm ERR! Additional logging details can be found in:
==> default: npm ERR!     /vagrant/workspace/gatekeeper/npm-debug.log
==> default: npm ERR! not ok code 0
==> default: npm ERR! network tunneling socket could not be established, cause=1
39978265110304:error:140770FC:SSL routines:SSL23_GET_SERVER_HELLO:unknown protoc
ol:../deps/openssl/openssl/ssl/s23_clnt.c:782:
==> default: npm ERR! network
==> default: npm ERR! network This is most likely not a problem with npm itself
==> default: npm ERR! network and is related to network connectivity.
==> default: npm ERR! network In most cases you are behind a proxy or have bad n
etwork settings.
==> default: npm ERR! network
==> default: npm ERR! network If you are behind a proxy, please make sure that t
he
==> default: npm ERR! network 'proxy' config is set properly.  See: 'npm help co
nfig'
```

Does anybody have a good idea where the following commands could be added to make it just work?

```
npm config set proxy (my proxy setting) --global
npm config set https-proxy (my proxy setting) --global
```

I'm working from a Windows 8.1 host using VirtualBox.

Best regards,
Martin

**Comment by GUI** [comment]
Sorry for the delay following up. I've also found NPM can be particularly finicky trying to get setup behind corporate proxies. I've run into this myself, so I might be able to dig up what I've done, but first off, are you trying to get API Umbrella running in Vagrant so you can do API Umbrella development? Or were you just wanting to use Vagrant as a simple way to spin up a copy of API Umbrella locally for playing/evaluation?

We definitely want to make setting up a development environment easy, but if you're just wanting to run a copy of API Umbrella to evaluate it first, the current Vagrant box is probably going to be overkill. It installs a lot of development dependencies and does further setup (like this NPM installation) that isn't needed if you just want to run a copy of API Umbrella inside VM. So if you're just looking to evaluate API Umbrella and not do development at this phase, let me know, because I've been thinking about setting up an alternative, simpler Vagrant box that could be used for those purposes (it should be much quicker and more straightforward, since in that case, all we'd need to do is install the single API Umbrella binary package). However, if you are looking to do active development, then we can try to figure out this NPM business.

**Comment by DonMartin76** [comment]
Hi @GUI, thanks for answering.

I have been struggling quite some over the course of the last couple of days, and this issue I ended up bribing our IT to take my computer into the list of  "computers which are allowed to bypass the proxy". In the end, I still couldn't get the vagrant machine up and running, for whichever reason.

Actually I did want to be able to develop things inside API Umbrella, and that's why I tried to set up the development environment. What has gotten me a little puzzled lately is the Lua branch you recently merged in. I have to read up on that and see if I can instead get that branch (the master so to say) up and running instead. Is there any documentation on how to do that?

/Martin
